Toggl.iOS/ViewSources/DateRangePickerViewSource.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Foundation;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using Toggl.Core.UI.Parameters;
+using Toggl.Core.UI.ViewModels.DateRangePicker;
+using Toggl.Core.UI.ViewModels.ReportsCalendar;
+using Toggl.iOS.Views;
+using UIKit;
+
+namespace Toggl.iOS.ViewSources
+{
+    public sealed class DateRangePickerViewSource : UICollectionViewSource
+    {
+        private const string cellIdentifier = nameof(ReportsCalendarViewCell);
+
+        private readonly UICollectionView collectionView;
+        private readonly ISubject<DateRangePickerMonthInfo> currentPageNotScrollingSubject = new Subject<DateRangePickerMonthInfo>();
+        private readonly ISubject<DateTime> dayTaps = new Subject<DateTime>();
+
+        private IImmutableList<DateRangePickerMonthInfo> months = ImmutableList<DateRangePickerMonthInfo>.Empty;
```

Comment:
```suggestion
        private ImmutableList<DateRangePickerMonthInfo> months = ImmutableList<DateRangePickerMonthInfo>.Empty;
```
The list is immutable and the concrete implementation is as safe to use as the interface. Even worse, the interface hides some members that the concrete implementation has (see another comment). That being said, the variable is also private, so you are not even hiding the impl details with this.